### ISSUE TITLE
Improve enum support in Predicate

### ISSUE THREAD

**Issue by victornoel** [hazelcast/hazelcast#8305]
Hi,

I noticed that when passing an enum to a `Predicate` such as `EqualPredicate`, the object was converted automatically by `EnumConverter` which would simply call `toString()` on it.

I am wondering if it's not better to use `name()` to have a more stable identifier for the enum.
Or even I'm wondering why the enum itself, which is already comparable, is not used…

I don't really have a solution for this question, @tombujok told me in the [mailing list](https://groups.google.com/d/msg/hazelcast/ZjMaxNMAzFU/A_UVV0OPAwAJ) that it would be complex to change that now, but it would be good to have an issue to track this question. So here it is :)

**Comment by victornoel** [comment]
See also #4910 maybe.

**Comment by mmedenjak** [comment]
Since we're heavily invested in the new query engine, I don't think we should spend too much time on this.

**Comment by mwashburn** [comment]
boooo

**Comment by mwashburn** [comment]
I agree that the functionality should not use a string, but for me the more urgent issue is it only works for some enums. 

When you have methods on the enum the instances of the  the enum are a subclass of the enum so class.isEnum returns false. 

https://stackoverflow.com/questions/61676052/classisenum-need-check-this-getsuperclass-java-lang-enum-class

ReflectionHelper#getAttributeType() should say Enum.class.isAssignableFrom(klass)

**Comment by mwashburn** [comment]
Same change is required in com.hazelcast.query.impl.predicates.AbstractPredicate#convertEnumValue
